lib/jsdom/living/helpers/style-rules.js

```diff
@@ -1,193 +1,40 @@
-    computedValue: "computed-color"
-  },
-  // https://drafts.csswg.org/css-ui-4/#propdef-outline-color
-  "outline-color": {
-    inherited: false,
-    initial: "invert",
-    computedValue: "computed-color"
-  },
-  // https://drafts.csswg.org/css-display/#the-display-properties
-  // Currently only "as-specified" is supported as a computed value
-  "display": {
-    inherited: false,
-    initial: "inline",
-    computedValue: "as-specified"
-  }
-};
-
-function forEachMatchingSheetRuleOfElement(elementImpl, handleRule) {
-  function handleSheet(sheet) {
-    forEach.call(sheet.cssRules, rule => {
-      if (rule.media) {
-        if (indexOf.call(rule.media, "screen") !== -1) {
-          forEach.call(rule.cssRules, innerRule => {
-            if (matches(innerRule, elementImpl)) {
-              handleRule(innerRule);
-            }
-          });
-        }
-      } else if (matches(rule, elementImpl)) {
-        handleRule(rule);
-      }
-    });
-  }
-
-  if (!parsedDefaultStyleSheet) {
-    parsedDefaultStyleSheet = cssom.parse(defaultStyleSheet);
-  }
-
-  handleSheet(parsedDefaultStyleSheet);
-  forEach.call(elementImpl._ownerDocument.styleSheets._list, handleSheet);
-}
-
-exports.invalidateStyleCache = elementImpl => {
-  if (elementImpl._attached) {
-    elementImpl._ownerDocument._styleCache = null;
-  }
-};
-
-exports.getDeclarationForElement = elementImpl => {
-  let styleCache = elementImpl._ownerDocument._styleCache;
-  if (!styleCache) {
-    styleCache = elementImpl._ownerDocument._styleCache = new WeakMap();
-  }
-
-  const cachedDeclaration = styleCache.get(elementImpl);
-  if (cachedDeclaration) {
-    return cachedDeclaration;
-  }
-
-  const declaration = new CSSStyleDeclaration();
-
-  function handleProperty(style, property) {
-    const value = style.getPropertyValue(property);
-    // https://drafts.csswg.org/css-cascade-4/#valdef-all-unset
-    if (value === "unset") {
-      declaration.removeProperty(property);
-    } else {
-      declaration.setProperty(
-        property,
-        value,
-        style.getPropertyPriority(property)
-      );
-    }
+const parsedDefaultStyleSheet = cssom.parse(defaultStyleSheet);
+const propertyList = new Map(Object.entries(cssstyle.propertyList));
+const initializableProperties = new Set([...propertyList].map(arr => {
+  const [key, { inherited, initial }] = arr;
+  if (!initial || initial.includes(" ")) {
+    return false;
+  } else if (!inherited || !/^(?:yes|no)$/i.test(inherited)) {
+    return false;
   }
-
-  forEachMatchingSheetRuleOfElement(elementImpl, rule => {
-    forEach.call(rule.style, property => {
-      handleProperty(rule.style, property);
-    });
-  });
-
-  forEach.call(elementImpl.style, property => {
-    handleProperty(elementImpl.style, property);
-  });
-
-  styleCache.set(elementImpl, declaration);
-
-  return declaration;
-};
+  return key;
+}).filter(key => key));
+const customProperties = new Map();
+const colorProperties = new Set();
```

Comment:
What are these global variables doing? It seems surprising to me that sharing custom properties and color properties across all jsdom instances would be correct. They should probably be document-scoped, if not element-scoped, right?